library/init/category/monad_fail.lean

```diff
@@ -4,12 +4,19 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Leonardo de Moura
 -/
 prelude
-import init.category.monad init.data.string.basic
+import init.category.lift init.data.string.basic
 
 universes u v
 
 class monad_fail (m : Type u → Type v) extends monad m :=
-(fail : Π {a}, string → m a)
+(fail {} : Π {a}, string → m a)
+
+-- deriving monad from monad_fail should happen only as a last resort
```

Comment:
why?

Reply:
I think I was mainly annoyed about the noise this instance introduced to the class_instance trace, since it's rarely the right thing to do. I can remove it.